### PR TITLE
Ck save load

### PR DESCRIPTION
**Pull request by Domily11**
Modified Save/Load function to take in backbone models. Currently working for models and config, but waiting for results to show up to test if the save load works for results as well.

### CODE REVIEW

File: leaf-ui/rappid-extensions/LinkInspector.js [hunk 1]
```diff
@@ -104,7 +104,8 @@ var LinkInspector = Backbone.View.extend({
     /** Chooses and sets correct template for link */
     render: function() {
         // Intention Link template
-        if(this.model.get('type') == 'element') {
+        console.log(this.model)
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/LinkInspector.js]: Delete.

File: leaf-ui/js/loadSaveFunctions.js [hunk 2]
```diff
@@ -198,21 +181,50 @@ function getFuncSegList(arr) {
...
  */
-function getModelJson() {
-	var obj = {};
-	obj.graph = graph.toJSON();
-	obj.model = model;
-	obj.analysisRequest = analysisRequest;
-	return obj;
+function createBBActor(cell){
+	var actor = cell.get('actor');
+	var actorbbm = new ActorBBM({type: actor.type, actorName: actor.attributes.actorName});
```
Comment (YesugenB) [inline leaf-ui/js/loadSaveFunctions.js]: Can we change the name of the var to maybe `actorBBM` ? All lowercase looks odd I think

File: leaf-ui/js/loadSaveFunctions.js [hunk 3]
```diff
@@ -198,21 +181,50 @@ function getFuncSegList(arr) {
...
+	cell.set('actor', actorbbm)
 }
 
+/**
+ * Returns a backbone model Link with information from the obj
+ *
+ */
+function createBBLink(cell){
+	var link = cell.get('link').attributes;
+	var linkbbm = new LinkBBM({displayType: link.displayType, linkType: link.linkType, postType: link.postType, absTime: link.absTime, evolving: link.evolving});
```
Comment (YesugenB) [inline leaf-ui/js/loadSaveFunctions.js]: Same for `linkbbm`? Think there aren't any `linkBBM`s

File: leaf-ui/js/onFunctions.js [hunk 4]
```diff
@@ -214,13 +184,11 @@ graph.on("add", function (cell) {
...
-            cell.prop("type", "Actor");
-            cell.label(0, { attrs: { text: { text: "is-a" } } });
-            cell.set('link', new LinkBBM({ linkType: 'is-a' }));
-        } else {
-            cell.prop("type", "element");
+	if (cell instanceof joint.shapes.basic.CellLink){
+        if (graph.getCell(cell.get("source").id) instanceof joint.shapes.basic.Actor){
+            cell.label(0,{attrs:{text:{text:"is-a"}}});
+            cell.set('link', new LinkBBM({displayType: "actor", linkType: 'is-a'}));
+		} else{
```
Comment (YesugenB) [inline leaf-ui/js/onFunctions.js]: Space after `else`

File: leaf-ui/css/analysis_html.css [hunk 5]
```diff
@@ -590,3 +590,28 @@ div.attribution{
...
+}
+
+#analysis-sidebar a{
+  padding: 16px;
+  display: block;
+}
+
+#analysis-sidebar a:hover {
+  background-color: #E8E8E8;
+}
```
Comment (Manarnz) [inline leaf-ui/css/analysis_html.css]: I tried to delete this from both files (analysis_HTML.css and analysis_style.css) and it worked just fine.

File: leaf-ui/js/loadSaveFunctions.js [hunk 6]
```diff
@@ -21,10 +21,9 @@ reader.onload = function() {
 	if (!reader.result) {
 		return;
 	}
-
 	var result = JSON.parse(reader.result);
 	loadFromObject(result);
-    var graphtext = JSON.stringify(graph.toJSON());
+    var graphtext = JSON.stringify(graph.toJSON())
```
Comment (Manarnz) [inline leaf-ui/js/loadSaveFunctions.js]: add ";" for consistency.

File: leaf-ui/js/loadSaveFunctions.js [hunk 7]
```diff
@@ -37,41 +36,24 @@ reader.onload = function() {
...
 		}
-	} else {
-		// Else if no analysisMap param, grab the analysisRequest
-		analysisRequest = Object.assign(new AnalysisRequest, obj.analysisRequest);
-		
 	}
-
-	graph.fromJSON(obj.graph);
+	// If the object contains configCollection, create configCollection fields from JSON
+	if (obj.configCollection != undefined) {
```
Comment (Manarnz) [inline leaf-ui/js/loadSaveFunctions.js:54]: What does the if statement exactly do?

File: leaf-ui/js/loadSaveFunctions.js [hunk 8]
```diff
@@ -198,21 +181,50 @@ function getFuncSegList(arr) {
 	return res;
 }
 
+*/
```
Comment (Manarnz) [inline leaf-ui/js/loadSaveFunctions.js:185]: Why are they commented out and not deleted?

File: leaf-ui/rappid-extensions/ActorInspector.js [hunk 9]
```diff
@@ -31,8 +31,10 @@ var ActorInspector = Backbone.View.extend({
          */
         render: function() {
             // If the clicked node is an actor, render the actor inspector
-            this.$el.html(_.template($(this.template).html())(this.actor.toJSON()));
+            console.log(this.model)
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/ActorInspector.js]: Delete it

**Review by Manarnz — COMMENTED** [review]
You did an amazing job Cat and all on your own. You are very intelligent. Good luck with your other PRs.

File: leaf-ui/js/loadSaveFunctions.js (hunk 7 above)
Comment (Domily11) [reply leaf-ui/js/loadSaveFunctions.js:54]: Try to determine if it needs to read the configCollection in

File: leaf-ui/js/loadSaveFunctions.js (hunk 8 above)
Comment (Domily11) [reply leaf-ui/js/loadSaveFunctions.js:185]: Later on we might want to take into account of the graphs made before the backbone-refactor, so I thought it was better to keep it.Could definitely move it though

**Review by YesugenB — APPROVED** [review]

File: leaf-ui/js/loadSaveFunctions.js (hunk 2 above)
Comment (Domily11) [reply leaf-ui/js/loadSaveFunctions.js]: Sure

File: leaf-ui/js/loadSaveFunctions.js (hunk 8 above)
Comment (Manarnz) [reply leaf-ui/js/loadSaveFunctions.js:185]: Maybe add a TODO so other people know why you kept them.

File: leaf-ui/rappid-extensions/BackboneModelsOther.js [hunk 10]
```diff
@@ -31,7 +31,8 @@
     },
 
     defaults: {
-        linkType: 'and',
+        displayType: "element",
```
Comment (YesugenB) [inline leaf-ui/rappid-extensions/BackboneModelsOther.js]: Is there a difference between using `""` and `''` here? If not could change so it is consistent with the other quotation marks

**Review by YesugenB — APPROVED** [review]
Looks good! The save/load works great and thanks for the fixes :)) Just left a small comment.

File: leaf-ui/js/loadSaveFunctions.js (hunk 8 above)
Comment (amgrubb) [reply leaf-ui/js/loadSaveFunctions.js:185]: I added the comment.

File: leaf-ui/rappid-extensions/BackboneModelsOther.js (hunk 10 above)
Comment (amgrubb) [reply leaf-ui/rappid-extensions/BackboneModelsOther.js]: Made consistent.

**Comment by amgrubb** [comment]
Great job Kat! Looking forward to round 2.

**Pull request merged**